static/src/javascripts/projects/common/modules/commercial/sticky-ad-banner.js

```diff
@@ -124,11 +124,12 @@ define([
         var transitionTimingFunction = 'cubic-bezier(0, 0, 0, .985)';
         var transitionDuration = '1s';
 
-        els.$header.css({
+        els.$page.css({
+            'will-change': 'transform',
```

Comment:
My understanding - and this is possibly rather dated - is that `will-change` essentially tells a browser that you want to promote an a node to its own compositor layer, reserving GPU memory for that specific purpose. There might be other consequences - turning off certain optimizations, or caching useful properties for faster reuse - but at a very high level, `will-change` is an alias for the old `translate3d` hack.

Thinking in terms of compositor memory is informative. Obviously, memory is a bounded resource, particularly scarce on mobile, so a complex page with lots of animations (i.e. several adverts) could quite feasibly exhaust it.

What happens when the compositor runs out of memory? I can think of four consequences:

1. If the browser cannot de-promote a `will-change` element from a compositor layer in a graceful, timely manner, it might throw an exception and crash;
2. If the browser is being asked to promote a new element with `will-change`, it might arbitrarily de-promote a `will-change` element that's older;
3. If the browser is being asked to promote a new element with `will-change`, it might just ignore the request and queue it for later consideration.
4. The browser bails out to software composition and possibly starts using a pagefile (yuk!)

The upshot is that we shouldn't add `will-change` arbitrarily, and that setting it too early consumes resources we'd really rather the browser allocated. At the same time, we don't want to set it too _late_. Promoting an element costs time as well as resource, and if we start the animation before promotion has completed, we incur that cost _and_ our animation remains unoptimized. This may well be worse than not using `will-change` at all.

Vendors are coy about the exact 'sweet spot' between setting the property too early and too late. Opera provide a little advice [here](https://dev.opera.com/articles/css-will-change-property/), suggesting 200ms might be the kind of gap to aim for, so it makes sense to assume a `will-change` optimization can take at least entire frames.